trunk/SUAVE/Methods/Propulsion/serial_HTS_dynamo_turboelectric_sizing.py

```diff
@@ -0,0 +1,284 @@
+    conditions_sls.freestream.density                     = np.atleast_1d(rho)
+    conditions_sls.freestream.dynamic_viscosity           = np.atleast_1d(mu)
+    conditions_sls.freestream.gravity                     = np.atleast_1d(planet.sea_level_gravity)
+    conditions_sls.freestream.isentropic_expansion_factor = np.atleast_1d(ducted_fan.working_fluid.compute_gamma(T,p))
+    conditions_sls.freestream.Cp                          = np.atleast_1d(ducted_fan.working_fluid.compute_cp(T,p))
+    conditions_sls.freestream.R                           = np.atleast_1d(ducted_fan.working_fluid.gas_specific_constant)
+    conditions_sls.freestream.speed_of_sound              = np.atleast_1d(a)
+    conditions_sls.freestream.velocity                    = conditions_sls.freestream.mach_number * conditions_sls.freestream.speed_of_sound
+    
+    # propulsion conditions
+    conditions_sls.propulsion.throttle           =  np.atleast_1d(1.0)    
+    
+    state_sls            = Data()
+    state_sls.numerics   = Data()
+    state_sls.conditions = conditions_sls   
+    results_sls          = ducted_fan.evaluate_thrust(state_sls)
+    
+    Turboelectric_HTS_Dynamo_Ducted_Fan.sealevel_static_thrust = results_sls.thrust_force_vector[0,0] / number_of_engines
+
+    # The shaft power is now known.
+    # To size the turboelectric generators the total required powertrain power is required.
+    # This is the sum of all the component s unpacked at the start.
+    # Each component sizing depends on the downstream component size.
+    # There are two streams: the main powertrain stream, and the cryogenic HTS rotor stream.
+    # Power conditioning out of the generator is considered as part of the turboelectric component.
+    # The two streams only join at this final stage.
+
+    # Get total power required by the main powertrain stream by applying power loss of each component in sequence
+    # Each component is considered as one instance, i.e. one engine
+    motor_input_power           = shaft_power/(motor.motor_efficiency * motor.gearbox_efficiency)
+    esc_input_power             = esc.power(motor_current, motor_input_power)
+    drive_power                 = esc_input_power
+
+    # Get power required by the cryogenic rotor stream
+    # The sizing conditions here are ground level conditions as this is highest cryocooler demand
+    HTS_current                 = rotor.current
+    rotor_input_power           = rotor.power(HTS_current, rotor.skin_temp)
+
+
+    # -------------- Current Supply Dynamo ---------------
+    # Here the HTS dynamo could be sized, however for now this just calculates the heating and power used by the basic dynamo as there is no sizing required for the basic dynamo model that has constant performance
+    # Get the shaft power required by the hts dynamo, and the heat load produced by the dynamo
+    dynamo_powers               = hts_dynamo.shaft_power(cryo_cold_temp, HTS_current, rotor_input_power)
```

Comment:
I'm starting to release many of the components don't follow the typical style.

Reply:
What could be done to resolve this?

Reply:
Take a look at other SUAVE [energy components.](https://github.com/suavecode/SUAVE/blob/develop/trunk/SUAVE/Components/Energy/Converters/Turbine.py#L59) They have inputs and outputs data structures. Then the only variable passed to the function is conditions.

Unfortunately this means you need to go file by file that you've been working with and put it in this format. But we did this for a reason.

Reply:
Would it work to move some of the functions to the Methods folder? 
Then modify the remaining functions to accept (self, conditions) as an input.

Reply:
Sure you can move things to the methods folder if that's easier.